### PR TITLE
Enable completed/pending/failed queue. Logic to extend worker events. Graceful RMQ exits

### PR DESCRIPTION
**Pull request by MoustafaEid**

### CODE REVIEW

File: packages/server/src/common/services/queue/classes/processor-base.ts [hunk 1]
```diff
@@ -18,7 +18,21 @@ export abstract class ProcessorBase<T extends WorkOrchestrator = WorkOrchestrato
...
+  async onFail(job: any, error): Promise<any> {
+    // console.log("FAIL");
+  }
+
+  async onRetry(job: any, error): Promise<any> {
+    // console.log("RETRY");
+  }
+
+  async onModuleDestroy() {
+    console.log("RMQ: ProcessorBase Shutdown");
```
Comment (mcharawi) [inline packages/server/src/common/services/queue/classes/processor-base.ts]: Can we use the standard logging here? `this.debug` etc?

File: packages/server/src/common/services/queue/classes/rmq-channel.ts [hunk 2]
```diff
@@ -11,10 +11,14 @@ export class RMQChannel {
   }
 
   async init() {
+    console.log("RMQ: Opening Channel");
```
Comment (mcharawi) [inline packages/server/src/common/services/queue/classes/rmq-channel.ts]: See above comment.

File: packages/server/src/common/services/queue/classes/rmq-channel.ts [hunk 3]
```diff
@@ -11,10 +11,14 @@ export class RMQChannel {
   }
 
   async init() {
+    console.log("RMQ: Opening Channel");
+
     this.channel = await this.connection.connection.createChannel();
   }
 
   async close() {
+    console.log("RMQ: Closing Channel");
```
Comment (mcharawi) [inline packages/server/src/common/services/queue/classes/rmq-channel.ts]: See above comment.

File: packages/server/src/common/services/queue/classes/rmq-connection-manager.ts [hunk 4]
```diff
@@ -38,6 +38,7 @@ export class RMQConnectionManager {
   }
 
 	async close(): Promise<void> {
+    console.log("Closing RMQConnectionManager")
```
Comment (mcharawi) [inline packages/server/src/common/services/queue/classes/rmq-connection-manager.ts]: See above comment.

File: packages/server/src/common/services/queue/classes/rmq-connection.ts [hunk 5]
```diff
@@ -12,6 +12,8 @@ export class RMQConnection {
     // TODO: stop primary process from connecting
     // since it doesn't do any actual work
     
+    console.log("RMQ: Opening Connection");
```
Comment (mcharawi) [inline packages/server/src/common/services/queue/classes/rmq-connection.ts]: See above comment.

File: packages/server/src/common/services/queue/classes/rmq-connection.ts [hunk 6]
```diff
@@ -27,6 +29,7 @@ export class RMQConnection {
   }
   
   async close() {
+    console.log("RMQ: Closing Connection");
```
Comment (mcharawi) [inline packages/server/src/common/services/queue/classes/rmq-connection.ts]: See above comment.

File: packages/server/src/common/services/queue/classes/work-orchestrator.ts [hunk 7]
```diff
@@ -1,56 +1,68 @@
...
-    });
+    channel.consume(
+      this.queueName,
+      consumeHandler,
+      options,
+    );
   }
 
   close(): Promise<void> {
+    console.log("RMQ: WorkOrchestrator Closing");
```
Comment (mcharawi) [inline packages/server/src/common/services/queue/classes/work-orchestrator.ts]: See above comment.

File: packages/server/src/common/services/queue/queue.explorer.ts [hunk 8]
```diff
@@ -37,6 +43,17 @@ export class QueueExplorer implements OnModuleInit {
     }
   }
 
+  async onModuleDestroy() {
+    Promise.all([
+      QueueManager.close(),
+      Producer.close()
+    ]);
+
+    if (process.env.LAUDSPEAKER_PROCESS_TYPE == 'QUEUE') {
```
Comment (mcharawi) [inline packages/server/src/common/services/queue/queue.explorer.ts]: Why is this conditional empty?

**Review by mcharawi — CHANGES_REQUESTED** [review]
Looks good, just need to standardize the logging

File: packages/server/src/common/services/queue/queue.explorer.ts (hunk 8 above)
Comment (MoustafaEid) [reply packages/server/src/common/services/queue/queue.explorer.ts]: The workers have their own shutdown handler that takes care of disconnecting on shutdown. I'll take this out

**Pull request merged**